### PR TITLE
Add circuitbreaker app to app store

### PR DESCRIPTION
**Pull request by aphex3k**
Initial PR for https://github.com/lightningequipment/circuitbreaker/issues/42

TODO:

1. @joostjager please review `umbrel-app.yml` and also replace `<tbd>` under `support: `.
2. @joostjager please review `docker-compose.yml`.
3. The line `submission: https://github.com/getumbrel/umbrel-apps/pull/122` will need to be adjusted after we know which id the PR will get

In the meantime I will test the app on my test node.

### CODE REVIEW

**Comment by aphex3k** [comment]
> In the meantime I will test the app on my test node.

I have successfully tested the installation of the app in 
- [x] [Umbrel dev environment](https://github.com/getumbrel/umbrel-dev)
- [ ] [Umbrel OS on a Raspberry Pi 4](https://github.com/getumbrel/umbrel-os)
- [x] [Custom Umbrel install on Linux](https://github.com/getumbrel/umbrel#-installation)

image

File: circuitbreaker/umbrel-app.yml [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+description: >-
+  It allows nodes to protect themselves from being flooded with htlcs. 
+  With circuitbreaker a maximum to the number of in-flight htlcs can be set on a per-peer basis.
+  Known and trusted peers for example can be assigned a higher maximum, while a new channel from a previously unseen node may be limited to only a few pending htlcs.
+  Furthermore it is possible to apply rate limits to the number of forwarded htlcs. 
+  This offers protection against DoS/spam attacks that rely on large numbers of fast-resolving htlcs. 
+  Rate limiting is implemented with a Token bucket. 
+  Via the UI the minimum interval between htlcs can be specified.
+  Large numbers of htlcs are also required for probing channel balances. 
+  Reducing the information leakage through probing could be another reason to put in place a rate limit for untrusted peers.
```
Comment (joostjager) [inline circuitbreaker/umbrel-app.yml:16]: Nice summary of the readme, looks sufficient.

File: circuitbreaker/umbrel-app.yml [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+  - lightning
+repo: https://github.com/lightningequipment/circuitbreaker
+support: <tbd>
+port: 9235
+gallery:
+  - 1.jpg
+  - 2.jpg
+  - 3.jpg
+  - 4.jpg
+  - 5.jpg
```
Comment (joostjager) [inline circuitbreaker/umbrel-app.yml]: Are so many images required? There really is only a single screen in circuitbreaker, so not sure how to make multiple screenshots out of that that are still adding value.

**Review by joostjager — COMMENTED** [review]
Thanks for setting this up, I really appreciate it!

File: circuitbreaker/umbrel-app.yml [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+  
+releaseNotes: >-
+  - Initial release on the umbrel app store
+  
+developer: Joost Jager
+website: https://github.com/joostjager
+dependencies: 
+  - lightning
+repo: https://github.com/lightningequipment/circuitbreaker
+support: <tbd>
```
Comment (joostjager) [inline circuitbreaker/umbrel-app.yml]: I think we can just link https://github.com/lightningequipment/circuitbreaker/issues for now?

**Comment by joostjager** [comment]
The coming week, I'll see if I can get a visual person to create an icon

File: circuitbreaker/umbrel-app.yml (hunk 2 above)
Comment (aphex3k) [reply circuitbreaker/umbrel-app.yml]: I believe the minimum they ask for is 3, the max is 5.

Comment (joostjager) [reply circuitbreaker/umbrel-app.yml]: Hmm yes, so what can we do here?

**Comment by joostjager** [comment]
@aphex3k here is the svg icon: https://svgur.com/s/q0r

File: circuitbreaker/docker-compose.yml [hunk 4]
```diff
@@ -0,0 +1,17 @@
+version: "3.7"
+
+services:
+  app_proxy:
+    environment:
+      APP_HOST: circuitbreaker_web_1
+      APP_PORT: 9235
+      
+  web:
+    image: ghcr.io/lightningequipment/circuitbreaker:v0.4.0@sha256:27d1a70ecacb15ff827ea1eca1fee39b7761d3ee602cdb169aec339e756daa98
```
Comment (joostjager) [inline circuitbreaker/docker-compose.yml]: I've tagged v0.4.1 which contains the icon as favicon.

Comment (aphex3k) [reply circuitbreaker/docker-compose.yml]: Excellent, I will update the PR.

File: circuitbreaker/umbrel-app.yml (hunk 2 above)
Comment (aphex3k) [reply circuitbreaker/umbrel-app.yml]: 1st one shows the main screen, # 2 could be a close up of the menu for default limits and # 3 could be a close up of per-channel configuration. Something like this?

image

image

Comment (joostjager) [reply circuitbreaker/umbrel-app.yml]: I guess that's indeed a way to generate unique screenshots :) Do you want to take them and add?

**Comment by aphex3k** [comment]
@joostjager I would start with these 3 and then let them give feedback.

**Comment by aphex3k** [comment]
If you are fine with the PR, please go ahead and merge it to the umbrel-apps fork and then I can use the fork to create the official PR in the umbrel repo.

**Comment by joostjager** [comment]
I've created https://github.com/getumbrel/umbrel-apps/pull/393. Thought it shouldn't be necessary to merge to the fork's master?

I think just the screenshots need to be added?

**Comment by joostjager** [comment]
Actually, I closed that PR above again so that you can open it directly on the umbrel repo and become the PR owner?

**Comment by aphex3k** [comment]
> Actually, I closed that PR above again so that you can open it directly on the umbrel repo and become the PR owner?

I can take care of it

**Comment by aphex3k** [comment]
https://github.com/getumbrel/umbrel-apps/pull/394

**Comment by joostjager** [comment]
I think that `Merge branch` commit isn't supposed to be there?

**Comment by aphex3k** [comment]
> I think that `Merge branch` commit isn't supposed to be there?

Fixed it.

**Comment by joostjager** [comment]
I think we can close this 'working' PR now?

**Pull request closed without merging**